python/tskit/vcf.py

```diff
@@ -115,13 +111,34 @@ def __init__(
                 sample_mask = np.array(sample_mask, dtype=bool)
                 self.sample_mask = lambda _: sample_mask
 
-    def __make_sample_mapping(self, ploidy):
+    def __make_sample_mapping(self, ploidy, individuals):
         """
         Compute the sample IDs for each VCF individual and the template for
         writing out genotypes.
         """
+        ts = self.tree_sequence
         self.samples = None
         self.individual_ploidies = []
+
+        if individuals is None:
+            # Find all sample nodes that reference individuals
+            self.individuals = np.unique(ts.nodes_individual[ts.samples()])
+            if len(self.individuals) == 1 and self.individuals[0] == tskit.NULL:
+                # No samples refer to individuals
+                self.individuals = []
```

Comment:
I don't see anything wrong here, but it seems confusing that if someone passes `individuals = None` then later in the code we can end up with `individuals = []`, but the user passing in `individuals = []` is forbidden. Maybe just leave `individuals=None` in this case?

Reply:
Good call - I'll try and make this cleaner.

Reply:
done